packages/cli/src/validate.ts

```diff
@@ -0,0 +1,221 @@
+import {warn, debug} from './console_utils'
```

Comment:
there's a lot of dupe logic from extractor here. Can you just extract & reuse?

Reply:
I will continue work on this PR when I next get chance